pkg/action/upgrade.go

```diff
@@ -426,5 +426,26 @@ func recreate(cfg *Configuration, resources kube.ResourceList) error {
 
 func objectKey(r *resource.Info) string {
 	gvk := r.Object.GetObjectKind().GroupVersionKind()
-	return fmt.Sprintf("%s/%s/%s/%s", gvk.GroupVersion().String(), gvk.Kind, r.Namespace, r.Name)
+
+	kind := gvk.Kind
+	groupVersion := gvk.GroupVersion().String()
+
+	if groupVersion == "apps/v1beta1" || groupVersion == "apps/v1beta2" {
+		groupVersion = "apps/v1"
```

Comment:
Why don't you just check `if old_api_version != new_api_version` instead?

Reply:
Good idea. Thanks for the hint. However we have to be a bit cautious about the Group. There might be the same Kind in different Groups.

For example:
NetworkPolicy - networking.k8s.io/v1
NetworkPolicy - crd.projectcalico.org/v1

In principle anyone could create a `Deplyoment` Kind in his CRD.

Reply:
Fair enough, but then you could check instead if only bit after "/" has been changed (for the same api type). That would be a more future proof solution - if there will be "apps/v2" version released in the future you'll have to rewrite this

Reply:
No, this will not be sufficient. As for example `Deployment` moved
`extensions/v1beta1` -> `apps/v1`

Please check the updated code :)

Reply:
Yeah, I see. Updated code looks cool tho +1